ui/src/plugins/dev.perfetto.ExplorePage/query_builder/widgets.scss

```diff
@@ -333,3 +333,169 @@
   flex-direction: column;
   gap: 8px;
 }
+
+// Inline edit list container - vertical stack with spacing
+.pf-inline-edit-list {
+  display: flex;
+  flex-direction: column;
+  gap: 8px;
+}
+
+// Form list item widget - inline form editing in a list item
+.pf-exp-form-list-item {
+  display: flex;
+  align-items: center;
+  gap: 8px;
+  padding: 4px 8px;
+  min-height: 40px;
+  background: var(--pf-color-background-secondary);
+  border: 1px solid var(--pf-color-border);
+  border-radius: 4px;
+  transition: all 0.15s ease-in-out;
+
+  &.pf-valid {
+    background: var(--pf-color-background-secondary);
+    border-color: var(--pf-color-border);
+
+    &:hover {
+      background: var(--pf-color-background);
+      border-color: var(--pf-color-primary);
+    }
+  }
+
+  &.pf-invalid {
+    background: rgba(255, 100, 100, 0.08);
+    border-color: rgba(255, 100, 100, 0.3);
+
+    &:hover {
+      background: rgba(255, 100, 100, 0.12);
+      border-color: rgba(255, 100, 100, 0.5);
+    }
+  }
+}
+
+.pf-exp-form-list-item-content {
+  flex: 1;
+  display: flex;
+  align-items: center;
+  gap: 8px;
+  min-width: 0;
+
+  // Ensure form controls don't overflow
+  .pf-select,
+  .pf-text-input {
+    min-width: 80px;
+  }
+}
+
+.pf-exp-form-list-item-actions {
+  display: flex;
+  align-items: center;
+  gap: 4px;
+  flex-shrink: 0;
+}
+
+// Material Design outlined input field with label on border
+.pf-outlined-field {
+  position: relative;
+  min-width: 120px;
+  flex: 1;
+  margin: 0;
+  padding: 0;
+  border: 1px solid var(--pf-color-border);
+  border-radius: 4px;
+  transition: border-color 0.15s ease-in-out;
+
+  &:hover:not(:disabled) {
+    border-color: var(--pf-color-text-secondary);
+  }
+
+  &:focus-within:not(:disabled) {
+    border-color: var(--pf-color-primary);
+    border-width: 2px;
```

Comment:
If you add a 1px wide box shadow instead of making the border thicker, this avoids the layout shift when focused